lib/fizzy/instantiate.hpp

```diff
@@ -117,11 +117,47 @@ struct ImportedFunction
     std::function<ExecutionResult(Instance&, const Value*, int depth)> function;
 };
 
-// Create vector of ExternalFunctions ready to be passed to instantiate.
-// imported_functions may be in any order,
-// but must contain functions for all of the imported function names defined in the module.
-std::vector<ExternalFunction> resolve_imported_functions(
-    const Module& module, std::vector<ImportedFunction> imported_functions);
+struct ImportedTable
+{
+    std::string module;
+    std::string name;
+    table_elements* table = nullptr;
+    uint32_t min = 0;
+    std::optional<uint32_t> max;
+};
+
+struct ImportedMemory
+{
+    std::string module;
+    std::string name;
+    bytes* data = nullptr;
+    uint32_t min = 0;
+    std::optional<uint32_t> max;
+};
+
+struct ImportedGlobal
+{
+    std::string module;
+    std::string name;
+    Value* value = nullptr;
+    ValType type = ValType::i32;
+    bool is_mutable = false;
+};
+
+struct Externals
```

Comment:
`instantiate` can also be changed to use `Externals` struct as input, but it can be another PR.